README.md

```diff
@@ -1182,7 +1186,226 @@ Messages sent have a priority based on their Service Type. Lower values for serv
+
+The error code in JSON data and the binary data are the same value from the same code list.
+
+##### 5.1.2.1 Payload
+
+The following query header is used by the system and the application to send error messages.
+
+<table width="100%">
+  <tr>
+    <th colspan="4">Binary Query Header</th>
+  <tr>
+  <tr>
+    <th>Query Type</th>
+    <th>TLS Message Type</th>
+    <th>Sequential Number</th>
+    <th>JSON Size</th>
+  </tr>
+  <tr>
+    <td>Notification</td>
+    <td>Send Internal Error</td>
+    <td>Any number to be used to correlate query messages</td>
+    <td>Size of the JSON data</td>
+  </tr>
+  <tr>
+    <td>0x20</td>
+    <td>0x000002</td>
+    <td>0xNNNNNNNN</td>
+    <td>0xNNNNNNNN</td>
+  </tr>
+  <tr>
+    <th colspan="4">JSON Data</th>
+  </tr>
+  <tr>
+    <th colspan="4">Binary Data: Single Byte Error Code</th>
+  </tr>
+</table>
+
+##### 5.1.2.2 JSON structure
+
+<table width="100%">
+  <tr>
+    <th>Key</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>id</td>
+    <td>A decimal value representing an Error code.</td>
+  </tr>
+  <tr>
+    <td>text</td>
+    <td>A string describing the error.</td>
+  </tr>
+</table>
+
+##### 5.1.2.3 Error codes
+
+<table width="100%">
+  <tr>
+    <th>Error code</th>
+    <th>Byte</th>
+    <th>Value</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>ERROR_SUCCESS</td>
+    <td>0x00</td><td>0</td>
+    <td>Internal Security Manager value</td>
+  </tr>
+  <tr>
+    <td>ERROR_INVALID_QUERY_SIZE</td>
+    <td>0x01</td><td>1</td>
+    <td>Wrong size of query data</td>
+  </tr>
+  <tr>
+    <td>ERROR_INVALID_QUERY_ID</td>
+    <td>0x02</td><td>2</td>
+    <td>Unknown Query ID</td>
+  </tr>
+  <tr>
+    <td>ERROR_NOT_SUPPORTED</td>
+    <td>0x03</td><td>3</td>
+    <td>SDL does not support encryption</td>
+  </tr>
+  <tr>
+    <td>ERROR_SERVICE_ALREADY_PROTECTED</td>
+    <td>0x04</td><td>4</td>
+    <td>Received request to protect a service that was protected before</td>
+  </tr>
+  <tr>
+    <td>ERROR_SERVICE_NOT_PROTECTED</td>
+    <td>0x05</td><td>5</td>
+    <td>Received handshake or encrypted data for not protected service</td>
+  </tr>
+  <tr>
+    <td>ERROR_DECRYPTION_FAILED</td>
+    <td>0x06</td><td>6</td>
+    <td>Decryption failed</td>
+  </tr>
+  <tr>
+    <td>ERROR_ENCRYPTION_FAILED</td>
+    <td>0x07</td><td>7</td>
+    <td>Encryption failed</td>
+  </tr>
+  <tr>
+    <td>ERROR_SSL_INVALID_DATA</td>
+    <td>0x08</td><td>8</td>
+    <td>SSL invalid data</td>
+  </tr>
+  <tr>
+    <td>ERROR_HANDSHAKE_FAILED</td>
+    <td>0x09</td><td>9</td>
+    <td>In case of all other handshake errors</td>
+  </tr>
+  <tr>
+    <td>INVALID_CERT</td>
+    <td>0x0A</td><td>10</td>
+    <td>Handshake failed because certificate is invalid</td>
+  </tr>
+  <tr>
+    <td>EXPIRED_CERT</td>
+    <td>0x0B</td><td>11</td>
+    <td>Handshake failed because certificate is expired</td>
+  </tr>
```

Comment:
@jacobkeeler looks like it is the values from older SYNC generations that are not used in newer systems. I removed these three to avoid changes in SDL core. https://github.com/smartdevicelink/protocol_spec/pull/41/commits/14c425acd8be165cc36d355bfb658c98b336240a